rct229/rulesets/ashrae9012019/section22/section22rule17.py

```diff
@@ -0,0 +1,128 @@
+
+        return {"heat_rejection_loop_ids_b": heat_rejection_loop_ids_b}
+
+    class HeatRejectionRule(PartialRuleDefinition):
+        def __init__(self):
+            super(Section22Rule17.HeatRejectionRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["loop", "rated_water_flowrate"],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            heat_rejection_b = context.baseline
+            heat_rejection_loop_ids_b = data["heat_rejection_loop_ids_b"]
+            heat_rejection_loop_b = heat_rejection_b["loop"]
+
+            return heat_rejection_loop_b in heat_rejection_loop_ids_b
+
+        def get_calc_vals(self, context, data=None):
+            heat_rejection_b = context.baseline
+
+            fan_shaft_power_b = (
+                heat_rejection_b["fan_shaft_power"]
+                if heat_rejection_b.get("fan_shaft_power")
+                else (
+                    getattr_(
+                        heat_rejection_b, "heat_rejections", "fan_motor_nameplate_power"
+                    )
+                    * FAN_SHAFT_POWER_FACTOR
+                    * getattr_(
+                        heat_rejection_b, "heat_rejections", "fan_motor_efficiency"
+                    )
+                )
+            )
+
+            rated_water_flowrate_b = heat_rejection_b["rated_water_flowrate"]
+            heat_rejection_efficiency_b = (
+                0.0
+                if fan_shaft_power_b == ZERO.POWER
+                else rated_water_flowrate_b / fan_shaft_power_b
+            )
+
+            return {
+                "fan_shaft_power_b": CalcQ("electric_power", fan_shaft_power_b),
+                "rated_water_flowrate_b": CalcQ(
+                    "volumetric_flow_rate", rated_water_flowrate_b
+                ),
+                "heat_rejection_efficiency_b": heat_rejection_efficiency_b,
+            }
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            heat_rejection_b = context.baseline
+
+            additional_note_for_no_shaft_power_b = (
+                ""
+                if heat_rejection_b.get("fan_shaft_power")
+                else (
+                    f"*Note: The fan shaft power for {heat_rejection_b['id']} was not given. For this evaluation, the fan shaft power was calculated using a rule of thumb "
+                    f"where fan_shaft_power = fan_motor_nameplate_power * {FAN_SHAFT_POWER_FACTOR} * fan_motor_efficiency."
+                )
+            )
+            heat_rejection_efficiency_b = calc_vals["heat_rejection_efficiency_b"]
+            heat_rejection_efficiency_in_gpm_per_hp_b = round(
+                heat_rejection_efficiency_b.to(ureg("gpm/hp")).magnitude, 1
+            )
+
+            if std_equal(HEAT_REJ_EFF_LIMIT, heat_rejection_efficiency_b):
+                UNDETERMINED_MSG = (
```

Comment:
convention is using a lower case variable name: `undetermined_msg`

Reply:
Addressed.